docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,262 @@
+| Standby T1-T0 link drop – Active        | Verify packet flow when T1-T0 link drop.                                 | • Start  sending traffic to active side.<br>• Configure DPU2 side T1-T0 link drop.              | DPU1 becomes standalone, DPU2 is standby.                                                | T2 receives packets without disruption.          |
+| Standby T1-T0 link drop - Standby       | Verify packet flow when T1-T0 link drop.                                 | • Start  sending traffic to standby side.<br>• Configure DPU2 side T1-T0 link drop.             | DPU1 becomes standalone, DPU2 is standby.                                                | T2 receives packets with allowed disruption.     |
+
+An example of CONFIG_DB ACL rule entry to drop NPU to local DPU probe packets will be
+```
+    "ACL_RULE": {
+        "ACL_LINK_DROP_TEST|LOCAL_PROBE_DROP1": {
+            "PACKET_ACTION": "DROP",
+            "PRIORITY": "1",
+            "SRC_IP": <NPU1 IP>,
+            "DST_IP": <DPU1 IP>,
+            "IP_TYPE": "IP",
+            "L4_SRC_PORT": "3784"
+        },
+        "ACL_LINK_DROP_TEST|LOCAL_PROBE_DROP2": {
+            "PACKET_ACTION": "DROP",
+            "PRIORITY": "1",
+            "SRC_IP": <DPU1 IP>,
+            "DST_IP": <NPU1 IP>,
+            "IP_TYPE": "IP",CO
+            "L4_SRC_PORT": "3784"
+        }
+     }
+```
+
+
+### Module 6 Critical Process Crash 
+For all process crash cases, we will have 4 variations, it’s 
+1. Process crash on DPU1, traffic landing on DPU2
+2. Process crash on DPU2, traffic landing on DPU1
+3. Process crash on DPU1, traffic landing on DPU1
+4. Process crash on DPU2, traffic landing on DPU2   
+
+The expected behavior is same, that HA state remains unchanged.
+
+| Case          | Goal                                     | Test Steps                                      | Expected Control Plane Behavior            | Expected Data Plane Behavior                                                                                                                                                               |
+| ------------- | ---------------------------------------- | ----------------------------------------------- | ------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| syncd on DPU  | Verify when syncd crash on DPU.          | • Start sending traffic<br>• Kill syncd on DPU  | DPU1 remains active, DPU2 remains standby. | T2 receives packets with allowed disruption.                                                                                                                                               |
```

Comment:
Please recheck and clarify expected behavior when syncd crashes. The syncd / swss containers usually restart and come up with empty config. Should we force a DPU reboot in this case?

Reply:
HI @mukeshmv - the point of these process crash test is to make sure the HA state can converge, and also measure the traffic impact. DPU reboot will be too generic.

Reply:
Also DPU reboot is included in `Module 7 Power down and hardware failure`